core/node/eth_sender/src/publish_criterion.rs

```diff
@@ -255,3 +261,86 @@ impl L1BatchPublishCriterion for L1GasCriterion {
         last_l1_batch
     }
 }
+
+struct GasConsts;
+
+struct CommitGasConsts {
+    base: u32,
+    per_batch: u32,
+}
+
+struct ExecuteCosts {
+    base: u32,
+    per_batch: u32,
+    per_l1_l2_tx: u32,
+}
+
+impl GasConsts {
+    /// Base gas cost of processing aggregated `Execute` operation.
+    /// It's applicable iff SL is Ethereum.
+    const AGGR_L1_BATCH_EXECUTE_BASE_COST: u32 = 200_000;
+    /// Base gas cost of processing aggregated `Execute` operation.
+    /// It's applicable if SL is  Gateway.
+    const AGGR_GATEWAT_BATCH_EXECUTE_BASE_COST: u32 = 300_000;
+
+    /// Base gas cost of processing aggregated `Commit` operation.
+    /// It's applicable if SL is Ethereum.
+    const AGGR_L1_BATCH_COMMIT_BASE_COST: u32 = 242_000;
+
+    /// Additional gas cost of processing `Commit` operation per batch.
+    /// It's applicable if SL is Ethereum.
+    const L1_BATCH_COMMIT_BASE_COST: u32 = 31_000;
+
+    /// Additional gas cost of processing `Commit` operation per batch.
+    /// It's applicable if SL is Gateway.
+    const AGGR_GATEWAY_BATCH_COMMIT_BASE_COST: u32 = 150_000;
+
+    /// Additional gas cost of processing `Commit` operation per batch.
+    /// It's applicable if SL is Gateway.
+    const GATEWAY_BATCH_COMMIT_BASE_COST: u32 = 200_000;
+
+    /// All gas cost of processing `PROVE` operation per batch.
+    /// It's applicable if SL is Ethereum.
+    const L1_BATCH_PROOF_GAS_COST_ETHEREUM: u32 = 500_000;
+
+    const GATEWAY_BATCH_EXECUTION_COST: u32 = 100_000;
+    const GATEWAY_L1_OPERATION_COST: u32 = 4_000;
+
+    fn commit_costs(is_gateway: bool) -> CommitGasConsts {
+        if is_gateway {
+            CommitGasConsts {
+                base: Self::GATEWAY_BATCH_COMMIT_BASE_COST,
+                per_batch: Self::AGGR_GATEWAY_BATCH_COMMIT_BASE_COST,
+            }
+        } else {
+            CommitGasConsts {
+                base: Self::L1_BATCH_COMMIT_BASE_COST,
+                per_batch: Self::AGGR_L1_BATCH_COMMIT_BASE_COST,
+            }
+        }
+    }
+
+    fn proof_costs(is_gateway: bool) -> u32 {
+        if is_gateway {
+            Self::L1_BATCH_PROOF_GAS_COST_ETHEREUM * 2
```

Comment:
I'd rather add a constant for this and put 1_000_000 in it. Also we shouldn't forget to update it for v28 (not critical)